ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -818,6 +803,20 @@ private static void createReplImportTasks(
         throw new SemanticException(ErrorMsg.DATABASE_NOT_EXISTS.getMsg(tblDesc.getDatabaseName()));
       }
     }
+
+    if (table != null) {
+      if (!replicationSpec.allowReplacementInto(table)) {
```

Comment:
Oh, just noticed, the caller pass different values for 2nd argument. So, will discuss with Sushanth and try to minimise the number of methods.